src/Form.js

```diff
@@ -14,16 +14,18 @@ class Form extends Component<PropsWithContext> {
   constructor(props: PropsWithContext) {
     super(props)
     if (!props._reduxForm) {
-      throw new Error(
-        'Form must be inside a component decorated with reduxForm()'
-      )
+      throw new Error('Form must be inside a component decorated with reduxForm()')
     }
   }
 
-  UNSAFE_componentWillMount() {
+  componentDidMount() {
     this.props._reduxForm.registerInnerOnSubmit(this.props.onSubmit)
   }
 
+  componentDidUpdate(nextProps: Props) {
+    this.props._reduxForm.registerInnerOnSubmit(nextProps.onSubmit)
```

Comment:
I think we should add a check here and register again only if onSubmit has changed

Reply:
Done